### DISCUSSION TITLE
A master admin site to monitor tenants

### DISCUSSION THREAD

**Discussion by VaZark** [django-tenants/django-tenants#754] [Q&A]
Is there any way to host a master admin site that is not tenant, exclusively so that we can manage the tenants? Or must all sites definitely be a client?

Being able to work off the public schema should technically be sufficient. It doesn't really need access to any tenant-specific apps.

**Comment by SilverFoxA** [comment]
@VaZark Did you find an optimal solution for this use-case?

**Reply by VaZark** [reply]
I ended up building my own tenancy solution that is based on separate database connections instead of different schemas on the same connection

**Comment by sgtsoft** [comment]
> Is there any way to host a master admin site that is not tenant, exclusively so that we can manage the tenants? Or must all sites definitely be a client?
> 
> Being able to work off the public schema should technically be sufficient. It doesn't really need access to any tenant-specific apps.

You can use a Mixin to display the Models in public schema and this way you can use the public schema as master admin site.

This is admin.py where I register the Domain and Client models. Though the APP is registered in both PUBLIC and TENANT schema, the Domain and Client models are seen only in public domain whose site admin is master site administrator.
```

class TenantSiteAdminOnlyMixin:
    """
    To allow access to Tenant and Domain models only to the super admin of the public site
    www.my-domain.com / localhost.com user
    To maintain the Tenants, Domains and Subscription models, which the tenants should not see
    """
    def _only_public_tenant_access(self, request):
        return True if request.tenant.schema_name == get_public_schema_name() else False

    def has_view_permission(self,request, view=None):
            return self._only_public_tenant_access( request)
    
    def has_add_permission(self,request, view=None):
            return self._only_public_tenant_access(request)
    
    def has_change_permission(self,request, view=None):
            return self._only_public_tenant_access( request)
    
    def has_delete_permission(self,request, view=None):
            return self._only_public_tenant_access( request)
    
    def has_view_or_change_permission(self,request, view=None):
            return self._only_public_tenant_access( request)        



@admin.register(Client)
class ClientAdmin(TenantSiteAdminOnlyMixin, TenantAdminMixin, admin.ModelAdmin):
    list_display = ['name', 'paid_until', 'on_trial', 'created_on', ]

@admin.register(Domain)
class DomainAdmin(TenantSiteAdminOnlyMixin, TenantAdminMixin, admin.ModelAdmin):
    list_display = ['domain', ]
```

**Comment by tomturner** [comment]
I personally wrote my own in the public area. I am able to create tenants etc.  It was quite easy to write.
